### PR TITLE
solution homework hb-06

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "left join fetch sc.user"
+                    + " where sc.user.id = :id", ShoppingCart.class);
+            query.setParameter("id", user.getId());
+            ShoppingCart cart = query.uniqueResult();
+            return cart;
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]:
```suggestion
            return query.getSingleResult();;
```

File: src/main/java/com/cinema/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -31,6 +35,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        User userDB = userService.add(user);
+        cartService.registerNewShoppingCart(userDB);
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
        userService.add(user);
        cartService.registerNewShoppingCart(user);
```

File: src/main/java/com/cinema/service/ShoppingCartService.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+
+public interface ShoppingCartService {
+    /**
+     * This method is responsible to add a Ticket to the ShoppingCart
+     *
+     * @param movieSession Contains the information required for a ticket
+     * @param user         - user who wan't to buy a ticket for a specific MovieSession
+     */
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/service/ShoppingCartService.java]: I think you don't need this

File: src/main/java/com/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        Ticket ticketBD = ticketDao.add(ticket);
+        ShoppingCart byUser = cartDao.getByUser(user);
```
Comment (MaliukDaria) [inline src/main/java/com/cinema/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        ShoppingCart userShoppingCart = cartDao.getByUser(user);
```

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.saveOrUpdate(shoppingCart);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: why save or update?

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "left join fetch sc.user"
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: we need left join for tickets, but not for user, for it's more logical to user inner

File: src/main/java/com/cinema/model/ShoppingCart.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@Entity
+@Table(name = "shopping_cart")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/ShoppingCart.java]: we don't need generatedValue if we use MapsId

Comment (RomanHreits) [reply src/main/java/com/cinema/model/ShoppingCart.java]: done

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 6 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: done

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 5 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**